src/app/public/testing/modal-fixture.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  ComponentFixture
+} from '@angular/core/testing';
+
+/**
+ * Allows interaction with a SKY UX modal component.
+ */
+export class SkyModalFixture {
+
+  private modalElement: HTMLElement;
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this.modalElement = document.querySelector('sky-modal[data-sky-id="' + skyTestId + '"]');
+
+    if (!this.modalElement) {
+      throw new Error(`No element was found with a \`data-sky-id\` value of "${skyTestId}".`);
+    }
+  }
+
+  /**
+   * The modal component's ARIA describedby attribute.
+   */
+  public get ariaDescribedBy(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('aria-describedby') || undefined;
```

Comment:
Should we have some null checks on `modalElement` before we try to access its attributes? The same goes for a few of the methods here.

Reply:
Added checks throughout.

Reply:
Do you think returning an `undefined` value is as valuable as alerting the consumer that the modal isn't found? Would it be more helpful if the test threw an error when the `modalElement` is null?